### PR TITLE
Update SA1110 to also check the parameter list in primary constructors

### LINKED ISSUE #3784: SA1110 does not trigger on primary constructors
I expect the following code to trigger SA1110 (opening parenthesis must be on declaration line), but it does not:
```
public record MyType
    (int X, int y)
{
}
```

Tested with version 1.2.0-beta.556

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #3784

### CODE REVIEW

**Comment by bjornhellander** [comment]
Any comments on this, @sharwell?

**Review by sharwell — APPROVED** [review]

File: StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs [hunk 1]
```diff
@@ -3,9 +3,66 @@
...
+        {
+            var testCode = $@"
+{typeKeyword} Foo
+    {{|#0:(|}}int x)
+{{
+}}";
+
+            var fixedCode = $@"
+{typeKeyword} Foo(
+    int x)
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs:51]: 💡 Can you also add tests for inheritance cases?

Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs:51]: Just record declarations that inherit from another record? Could you describe some more what you are after?

Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs:51]: Yes, reference type records and classes with primary constructors allow a base type to be specified, and arguments can be provided to the base class.

```cs
record BaseRecord(int A);

record DerivedRecord(int A, int B) : BaseRecord(A);
```

```cs
class BaseClass(int a);

class DerivedClass(int a, int b) : BaseClass(int a);
```

Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs:51]: So the tests that you wanted was regarding the placement of the opening parenthesis **in the call to the base constructor**?

Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs:51]: Yes, that's correct. I believe SA1110 handles this for the equivalent `base(...)` call in constructors.

Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs:51]: Fixed

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/Helpers/CommonMemberData.cs [hunk 2]
```diff
@@ -137,5 +137,13 @@ public static IEnumerable<object[]> TypeKeywordsWhichSupportPrimaryConstructors
                 }
             }
         }
+
+        public static IEnumerable<object[]> ReferenceTypeKeywordsWhichSupportPrimaryConstructors
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/Helpers/CommonMemberData.cs:141]: 💡 Consider rewriting this the other way around so `TypeKeywordsWhichSupportPrimaryConstructors` calls `ReferenceTypeKeywordsWhichSupportPrimaryConstructors`.

**Review by sharwell — APPROVED** [review]

**Pull request merged**